server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -735,97 +734,101 @@ public int compare(File o1, File o2)
           }
       );
       boolean isCorrupted = false;
-      try {
```

Comment:
this was catching the IOException which could only come from announcer inside this moved this inside the addSink method, other IOEx inside this try/catch were already handled by the inner catch blocks.